components/ActivityCard.js

```diff
@@ -1,34 +1,43 @@
 import Image from "next/image";
+import Link from "next/link";
 import styled from "styled-components";
+import { useRouter } from "next/router";
 
-export default function ActivityCard({ title, categories, imageUrl }) {
-  return (
-    <StyledCardSection>
-      <StyledImageDiv>
+export default function ActivityCard({ id, title, categories, imageUrl }) {
+  const router = useRouter();
+
+
+  return ( 
+    <StyledArticle>
+    <StyledImageContainer>
         <Image
           src={imageUrl}
           alt={title}
           style={{ objectFit: "cover" }}
           sizes="33vw"
           fill
         />
-      </StyledImageDiv>
-      <StyledList>
-        {categories.map((category) => (
-          <StyledListItem key={category}>{category}</StyledListItem>
-        ))}
-      </StyledList>
-      <StyledTitle>{title}</StyledTitle>
-    </StyledCardSection>
-  );
+    </StyledImageContainer>
+    <StyledList>
+      {categories.map((category) => (
+        <StyledListItem key={category}>{category}</StyledListItem>
+      ))}
+    </StyledList>
+    <Link href={`/activity/${id}`}>
+    <StyledTitle>
+      {title}
+    </StyledTitle></Link>
```

Comment:
That's fine but now the Link is as wide as the card and not as wide as the title. You can solve this by making the title `display: inline;`.